UNI-36: Connect the chat client to our backend

derekxu04 opened the pull request:
Should be working - if you login, put in your faculty, you should be directed to the '/chats' page which should display a chat involving yourself (based on your userID and userToken) and someone else

Siyer2 commented on backend/.env.local:
```diff
@@ -1 +1,2 @@
-ENVIRONMENT=local
\ No newline at end of file
+ENVIRONMENT=local
+REACT_APP_STREAM_API_KEY=94ggnkahafdr
```
I think we need a better strategy for our env vars. Having a backend/.env.local and a backend/.env is confusing. I recommend we do the following:
1. Move all .env.local into .env (so can delete .env.local) - note: this might require a change to [app.ts](https://github.com/Siyer2/uni-connector/blob/c41d4eff58389d4753854219e368797f7472b975/backend/src/app.ts#L9)
2. Create a .env.example that is NOT gitignored so people will know what's expected. It can look something like this:
```
REACT_APP_STREAM_API_SECRET=example
ENVIRONMENT=local
REACT_APP_STREAM_API_KEY=example
```
3. Message in general that you need this .env file in backend for the project to continue working

Also, I think it's a good idea to change `REACT_APP_STREAM_API_KEY` to `REACT_APP_STREAM_PUBLIC_KEY`

derekxu04 replied:
where should I say that you need the .env file in the backend - in the README at the root of the project?

Siyer2 commented on backend/src/chats/index.ts:
```diff
@@ -0,0 +1,14 @@
+
+export function getChatUserToken(user_id: string): string {
+  // Define values.
+  const api_key = process.env.REACT_APP_STREAM_API_KEY;
+  const api_secret = process.env.REACT_APP_STREAM_API_SECRET;
```
We should use camelCase here not snake_case for the variable names

Siyer2 commented on backend/src/chats/index.ts:
```diff
@@ -0,0 +1,14 @@
+import { StreamChat } from 'stream-chat';
+
+export function getChatUserToken(user_id: string): string {
+  // Define values.
```
Sorry my OCD really wants you to remove the . at the end of this comment also 😅

Siyer2 commented on frontend/src/pages/Chats.tsx:
```diff
@@ -24,23 +25,28 @@ import './Chats.css';
 
-      const channel = getChannel(chatClient, user.id);
+      // Note that creating a channel using userIDs requires at least two
+      const channel = getChannel(chatClient, [user.primaryKey.slice(5), 'john'], user.name);
```
This is an interesting one. It seems that by default, users are put into a group with a fake user called 'john'. Rather than doing this, in `init`, is it possible to get all channels for a user. If channels.length === 0, then we can display a message that tells them to look out on Tuesday for their first match. 

If channels.length > 0, then we can setChannel to the last one

derekxu04 commented:
I think I've addressed all your comments.

When you say: 'message in general that you need this .env file in backend for the project to continue working' - where should I put that - in the README at the root of the project?

To see when `channels.length === 0` change `filters` in Chats.tsx to have a random `user.id`

Siyer2 reviewed (approved):
Awesome work! There's some code that's been commented out in [Chats.tsx](https://github.com/Siyer2/uni-connector/blob/e99fe103803f283e9442a492ee1df5f34437ea39/frontend/src/pages/Chats.tsx#L45) - remove those comments and then you can merge 🙌
